Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -1,127 +1,273 @@
-      :param initial: Initial guess ( Default initial = 0)
-      :param f: Differentiable function
-      :param g: Convex function with " simple " proximal operator
-
-
-    Reference:
-      
-        Beck, A. and Teboulle, M., 2009. A fast iterative shrinkage-thresholding 
-        algorithm for linear inverse problems. 
-        SIAM journal on imaging sciences,2(1), pp.183-202.
-    '''
-    
-    
-    def __init__(self, initial=None, f=None, g=ZeroFunction(), **kwargs):
-        
-        '''FISTA algorithm creator 
-        
-        initialisation can be done at creation time if all 
-        proper variables are passed or later with set_up
-        
-        Optional parameters:
 
-        :param initial: Initial guess ( Default initial = 0)
-        :param f: Differentiable function
-        :param g: Convex function with " simple " proximal operator'''
-        
-        super(FISTA, self).__init__(**kwargs)
-        if kwargs.get('x_init', None) is not None:
-            if initial is None:
-                warnings.warn('The use of the x_init parameter is deprecated and will be removed in following version. Use initial instead',
-                   DeprecationWarning, stacklevel=4)
-                initial = kwargs.get('x_init', None)
+class ISTA(Algorithm):
+
+    r"""Iterative Shrinkage-Thresholding Algorithm, see :cite:`BeckTeboulle_b`, :cite:`BeckTeboulle_a`.
+
+    Iterative Shrinkage-Thresholding Algorithm (ISTA)
+
+    .. math:: x^{k+1} = \mathrm{prox}_{\alpha^{k} g}(x^{k} - \alpha^{k}\nabla f(x^{k}))
+
+    is used to solve
+
+    .. math:: \min_{x} f(x) + g(x)
+
+    where :math:`f` is differentiable, :math:`g` has a *simple* proximal operator and :math:`\alpha^{k}`
+    is the :code:`step_size` per iteration.
+
+    Note
+    ----
+
+    For a constant step size, i.e., :math:`a^{k}=a` for :math:`k\geq1`, convergence of ISTA
+    is guaranteed if
+
+    .. math:: \alpha\in(0, \frac{2}{L}),
+
+    where :math:`L` is the Lipschitz constant of :math:`f`, see :cite:`CombettesValerie`.
+
+    Parameters
+    ----------
+
+    initial : DataContainer
+              Initial guess of ISTA.
+    f : Function
+        Differentiable function
+    g : Function
+        Convex function with *simple* proximal operator
+    step_size : positive :obj:`float`, default = None
+                Step size for the gradient step of ISTA.
+                The default :code:`step_size` is :math:`\frac{0.99 * 2}{L}.`
+    kwargs: Keyword arguments
+        Arguments from the base class :class:`.Algorithm`.
+
+
+    Examples
+    --------
+
+    .. math:: \underset{x}{\mathrm{argmin}}\|A x - b\|^{2}_{2}
+
+    >>> f = LeastSquares(A, b=b, c=0.5)
+    >>> g = ZeroFunction()
+    >>> ig = Aop.domain
+    >>> ista = ISTA(initial = ig.allocate(), f = f, g = g, max_iteration=10)
+    >>> ista.run()
+
+
+    See also
+    --------
+
+    :class:`.FISTA`
+    :class:`.GD`
+
+
+    """
+
+    @property
+    def provable_convergence_condition(self):
```

Comment:
As above I thought this was going to be
`_provable_convergence_condition()`


which is then exposed to the user through the inherited method:
`is_provably_convergent()`